hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/AbstractS3ACommitter.java

```diff
@@ -349,16 +368,9 @@ public void recoverTask(TaskAttemptContext taskContext) throws IOException {
    * @throws IOException IO failure
    */
   protected void maybeCreateSuccessMarkerFromCommits(JobContext context,
-      List<SinglePendingCommit> pending) throws IOException {
+      ActiveCommit pending) throws IOException {
     List<String> filenames = new ArrayList<>(pending.size());
-    for (SinglePendingCommit commit : pending) {
-      String key = commit.getDestinationKey();
-      if (!key.startsWith("/")) {
-        // fix up so that FS.makeQualified() sets up the path OK
-        key = "/" + key;
-      }
-      filenames.add(key);
-    }
+    filenames.addAll(pending.committedObjects);
```

Comment:
If this is not going to create the _SUCCESS marker because the file list is too large, why get all committed file names here? I think this should be inside whatever check `maybeCreateSuccessMarker` has to avoid the memory consumption.

Reply:
we stop adding files to that pending.committed objects list once we reach an arbitrary threshold (SUCCESS_MARKER_FILE_LIMIT == 100), so we add that subset of entries. I'll clarify that in the comments